PINCache/PINDiskCache.m

```diff
@@ -89,14 +89,10 @@ - (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath
         NSString *pathComponent = [[NSString alloc] initWithFormat:@"%@.%@", PINDiskCachePrefix, _name];
         _cacheURL = [NSURL fileURLWithPathComponents:@[ rootPath, pathComponent ]];
         
-        __weak PINDiskCache *weakSelf = self;
-        dispatch_async(_asyncQueue, ^{
-            PINDiskCache *strongSelf = weakSelf;
-            [strongSelf lock];
-                [strongSelf createCacheDirectory];
-                [strongSelf initializeDiskProperties];
-            [strongSelf unlock];
-        });
+        [self lock];
```

Comment:
Do you still need to lock this section while we're still in the initializer?  The lock (semaphore) is per-instance.